docs/src/modules/components/AppSearch.js

```diff
@@ -1,119 +1,28 @@
-          '.algolia-autocomplete.algolia-autocomplete': {
-            '& .ds-dropdown-menu': {
-              boxShadow: `0px 4px 20px ${
-                theme.palette.mode === 'dark'
-                  ? alpha(theme.palette.background.paper, 0.72)
-                  : 'rgba(170, 180, 190, 0.3)'
-              }`,
-              border: '1px solid',
-              borderColor:
-                theme.palette.mode === 'dark'
-                  ? theme.palette.primaryDark[400]
-                  : theme.palette.grey[200],
-              borderRadius: theme.shape.borderRadius,
-              '&::before': {
-                display: 'none',
-              },
-              '& [class^=ds-dataset-]': {
-                border: 0,
-                maxHeight: 'calc(100vh - 100px)',
-                borderRadius: theme.shape.borderRadius,
-                backgroundColor: theme.palette.background.paper,
-              },
-            },
-            '& .algolia-docsearch-suggestion--category-header-lvl0': {
-              color: theme.palette.text.primary,
-            },
-            '& .algolia-docsearch-suggestion .algolia-docsearch-suggestion--subcategory-column': {
-              opacity: 1,
-              padding: '5.33px 10.66px',
-              textAlign: 'right',
-              width: '25%',
-            },
-            '& .algolia-docsearch-suggestion .algolia-docsearch-suggestion--content': {
-              float: 'right',
-              padding: '5.33px 0 5.33px 10.66px',
-              width: '75%',
-            },
-            '& .algolia-docsearch-suggestion--subcategory-column-text': {
-              color: theme.palette.text.secondary,
-              fontWeight: theme.typography.fontWeightRegular,
-            },
-            '& .algolia-docsearch-suggestion--highlight': {
-              color: theme.palette.mode === 'light' ? '#174d8c' : '#acccf1',
-            },
-            '& .algolia-docsearch-suggestion': {
-              textDecoration: 'none',
-              backgroundColor: theme.palette.background.paper,
-            },
-            '& .algolia-docsearch-suggestion--title': {
-              ...theme.typography.h6,
-              color: theme.palette.text.primary,
-            },
-            '& .algolia-docsearch-suggestion--text': {
-              ...theme.typography.body2,
-              color: theme.palette.text.secondary,
-            },
-            '&& .algolia-docsearch-suggestion--no-results': {
-              width: '100%',
-              '&::before': {
-                display: 'none',
-              },
-            },
-            '& .ds-dropdown-menu .ds-suggestion.ds-cursor .algolia-docsearch-suggestion--content': {
-              backgroundColor: `${theme.palette.action.selected} !important`,
-            },
-          },
-        };
-      }}
-    />
-  );
+function isAppleDevice() {
+  if (typeof navigator !== 'undefined') {
+    return /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
+  }
+  return '';
 }
```

Comment:
This component shouldn't be rendered on the server. We used to lazy load the search on the client. If that's no longer the case, then that's a separate bug we need to fix before this change.

Reply:
This component is not used directly. There is a wrapper called `DeferredAppSearch` in `AppFrame.js`.

Reply:
> This component is not used directly. There is a wrapper called `DeferredAppSearch` in `AppFrame.js`.

I know that. That's what I was referring to.

Reply:
> If that's no longer the case, then that's a separate bug we need to fix before this change.

@eps1lon Can you explain a bit more? I am not what should be the next step.

Reply:
This code guards against being rendered on the server. But this component should not be rendered on the server. The question is not why you should fix it. The question is why did you change it?

Reply:
I see. I have moved inside the component same as before the change.